### ISSUE TITLE
Versions missmatch during build process (probably a priority issue ?)

### ISSUE THREAD

**Issue by MiMillieuh** [blue-build/template#35]
Hello, 

Since a bit more than a week, I get these issues randomly when building an image : 

```
 17.22   - package kernel-devel-matched-6.8.7-300.fc40.x86_64 from updates-archive requires kernel-core = 6.8.7-300.fc40, but none of the providers can be installed
17.22   - package kernel-devel-matched-6.8.8-300.fc40.x86_64 from updates-archive requires kernel-core = 6.8.8-300.fc40, but none of the providers can be installed
17.22   - package kernel-devel-matched-6.8.9-300.fc40.x86_64 from updates-archive requires kernel-core = 6.8.9-300.fc40, but none of the providers can be installed
17.22   - cannot install both kernel-core-6.9.5-200.fc40.x86_64 from updates and kernel-core-6.9.4-200.fc40.x86_64 from @System
17.22   - cannot install both kernel-core-6.8.5-301.fc40.x86_64 from fedora and kernel-core-6.9.4-200.fc40.x86_64 from @System
17.22   - cannot install both kernel-core-6.8.10-300.fc40.x86_64 from updates-archive and kernel-core-6.9.4-200.fc40.x86_64 from @System
17.22   - cannot install both kernel-core-6.8.7-300.fc40.x86_64 from updates-archive and kernel-core-6.9.4-200.fc40.x86_64 from @System
17.22   - cannot install both kernel-core-6.8.8-300.fc40.x86_64 from updates-archive and kernel-core-6.9.4-200.fc40.x86_64 from @System
17.22   - cannot install both kernel-core-6.8.9-300.fc40.x86_64 from updates-archive and kernel-core-6.9.4-200.fc40.x86_64 from @System
17.22   - conflicting requests
```

I say they comes randomly because it's possible to fix them manually by adding this to module : 
```
  - type: containerfile
    snippets:
      - RUN rpm-ostree override replace --experimental --from repo=updates libstdc++ libgomp libgcc cpp
```

But with the specific package that causes issue.

However if it works again, it will generate a new error, package that I get the most error are usually gcc related but in that example you can see that even kernel can cause issues.

The issue seems like it prioritize update-archive instead of update.

**Comment by fiftydinar** [comment]
I have this issue too.

This looks like the upstream image build issue rather than BlueBuild-related.

We don't manage anything regarding repos except when installing akmods.

In `akmods` module, we:
- Set the higher priority of 90 to ublue-os/akmods COPR repo (to install akmods 1st from there instead from fedora-update)
- Enable negativo17 multimedia repo for xpadneo akmod
- Install akmods
- Disable negativo17 multimedia repo (to prevent repo collision)

Otherwise, nothing regarding repos is modified.

**Comment by fiftydinar** [comment]
Build without akmods succeed

**Comment by MiMillieuh** [comment]
> Build without akmods succeed

Thanks for trying this,

I assume it's related to some package installation in the image like kernel-devel in the current case or gcc when it happens.

Honestly these issues happened without me changing anything to my image. that's strange cause one day it builds the other it doesn't and I haven't change or even done a commit between the two builds.

**Comment by fiftydinar** [comment]
> Honestly these issues happened without me changing anything to my image. that's strange cause one day it builds the other it doesn't and I haven't change or even done a commit between the two builds.

Yeah, it happens from time to time, as noted in BlueBuild `akmods` docs:
https://blue-build.org/reference/modules/akmods/#outdated-akmods-compared-to-the-current-kernel-version-fail-the-build

I will investigate if this process can be improved.

We currently copy what Bluefin & Bazzite images do regarding akmods inside `akmods` module, where process steps are mentioned in my previous comment.

But, this still looks mostly like the upstream issue.

If you notice something that improves this, please let me know.

**Comment by fiftydinar** [comment]
I will transfer this issue to bluebuild/modules, because I think it's more appropriate to be there.
